cli/command/image/build_session.go

```diff
@@ -53,7 +54,16 @@ func addDirToSession(session *session.Session, contextDir string, progressOutput
 
 	p := &sizeProgress{out: progressOutput, action: "Streaming build context to Docker daemon"}
 
-	workdirProvider := filesync.NewFSSyncProvider(contextDir, excludes)
+	workdirProvider := filesync.NewFSSyncProvider([]filesync.SyncedDir{
+		{Dir: contextDir},
+		{Excludes: excludes},
+		{Map: func(s *fsutil.Stat) bool {
```

Comment:
Shouldn't all of these be in a single `SyncedDir` instead of 3 separate ones?

I'm alsonot seeing a `Map` field in master,so we should wait for https://github.com/moby/buildkit/pull/126/ to merge and vendor in master instead of the PR branch.

Reply:
Yes.. only one you're right 😊 . And yes `Map` is part of https://github.com/moby/buildkit/pull/126 this PR needs to use master of course before merge.